external-crates/move/crates/move-unit-test/src/test_reporter.rs

```diff
@@ -110,6 +112,53 @@ impl FailureReason {
     }
 }
 
+fn clever_error_line_number_to_loc(test_plan: &TestPlan, vm_error: &VMError) -> Option<Loc> {
+    let abort_code = match (vm_error.major_status(), vm_error.sub_status()) {
+        (StatusCode::ABORTED, Some(abort_code)) => abort_code,
+        _ => return None,
+    };
+    let location = vm_error.location();
+    let bitset = ErrorBitset::from_u64(abort_code)?;
+    if bitset.identifier_index().is_some() || bitset.constant_index().is_some() {
+        return None;
+    }
+    let line_number = bitset.line_number()? - 1;
+
+    match location {
+        Location::Undefined => None,
+        Location::Module(module_id) => {
+            let source_map = &test_plan.module_info.get(module_id)?.source_map;
+            let file_hash = source_map.definition_location.file_hash();
+            let file_id = test_plan.mapped_files.file_mapping().get(&file_hash)?;
+            let line_range = test_plan
+                .mapped_files
+                .files()
+                .line_range(*file_id, line_number as usize)
+                .ok()?;
+
+            let mut start = line_range.start;
+            // Sub 1 to trim off newline character
+            let end = line_range.end - 1;
+
+            // Since we only have the line span, trim the start of the line to remove any
+            // whitespace.
+            let source_slice =
+                &test_plan.mapped_files.files().source(*file_id).unwrap()[start..end];
+            if let Some((whitespace, rest)) =
+                source_slice.split_once(|ch: char| !ch.is_ascii_whitespace())
```

Comment:
Would it be more efficient to call `trim_start()` and do the abs length comparison?

Reply:
It would indeed...great idea!